### PR TITLE
Show error when volatile flag is given

### PR DESCRIPTION
**Pull request by unnonouno**
I fixed Variable to show error message when obsolete `volatile` argument is given.

### CODE REVIEW

File: chainer/variable.py [hunk 1]
```diff
@@ -240,9 +241,18 @@ def _set_grad_with_check(self, g, func, var):
...
+        data, name, grad, initializer, update_rule, requires_grad):
+    return Variable(
+        data, name=name, grad=grad, initializer=initializer,
+        update_rule=update_rule, requires_grad=requires_grad)
+
+
 class Variable(object):
 
-    """Array with a structure to keep track of computation.
+    """__init__(data=None, name=None, grad=None, initializer=None, update_rule=None, requires_grad=True)
```
Comment (beam2d) [inline chainer/variable.py]: Could you add `*,` between `data=None,` and `name=None`, like `data=None, *, name=None, ...`? It is a syntax of Py3 that indicates the following arguments are keyword only, and would be informative for users to notice that arguments after `data` cannot be passed as positional ones.

Comment (unnonouno) [inline chainer/variable.py]: Ok. Do we need to fix other docstrings?

Comment (beam2d) [inline chainer/variable.py]: It might be better to fix them, too, in another PR.

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**